gitops documentation (#359)
razee-io/RemoteResource: gitops.md

> # Gitops

Reviewer: Git

> How to use Remote Resource gitops to deliver by:

Reviewer: ```suggestion
The RemoteResource operator git backendService supports delivering contents from both the GitHub and GitLab services by [branch](#branch), [tag](#tag-or-commit), [commit](#tag-or-commit), or [release artifact](#release).  
```

## 1. Branch (implemented)

Provide:

* repo url (use one of following formats)
  * `https://github.com/razee-io/RemoteResource.git`
  * `git@github.com:razee-io/RemoteResource.git`
* branch
* filePath
  * path/to/directory/*
  * path/to/directory/*.extension
  * path/to/directory/filename.extension
* personal access token (if not public)

**Github:**

Sample RR:

Use `git` as backend service. Provide inputs as git request option. Personal access token should be a secret.

```yaml
apiVersion: "deploy.razee.io/v1alpha2"
kind: RemoteResource
metadata:
  name: <remote_resource_name>
  namespace: <namespace>
spec:
  clusterAuth:
    impersonateUser: razeedeploy
  backendService: git
  requests:
    - options:
        git:
          provider: 'github'
          repo: "https://github.com/razee-io/RemoteResource.git"
          branch: "master"
          filePath: "*.yaml"
        headers:
          Authorization:
            valueFrom:
              secretKeyRef:
                name: token
                namespace: <namespace>
                key: token
```

> Api (how provided inputs get mapped to api behind the scenes):

Reviewer: ```suggestion
Implementation detail: (i.e. How provided inputs get mapped to api behind the scenes):
```

> 
> Api (how provided inputs get mapped to api behind the scenes):
> 
> * Get listing of files from branch/path provides raw download_url: `https://api.github.com/repos/{repo}/contents/{path}?ref={branch}`
> * request to download_url to get file

Reviewer: ```suggestion

Implementation detail: (i.e. How provided inputs get mapped to api behind the scenes):

- GET  `https://api.github.com/repos/{repo}/contents/{path}?ref={branch}` provides list of raw download_url(s) for file(s)
- request to download_url to get file
```
